ros2pkg/ros2pkg/api/create.py

```diff
@@ -63,19 +106,90 @@ def _create_folder(folder_name, base_directory, exist_ok=True):
     return folder_path
 
 
-def _create_template_file(template_file_name, output_directory, output_file_name, template_config):
-    template_path = pkg_resources.resource_filename('ros2pkg', 'resource/' + template_file_name)
-    if not os.path.exists(template_path):
-        raise FileNotFoundError('template not found:', template_path)
+def create_package_basics(args,
+                          buildtool_depends=None,
+                          test_dependencies=[]):
+    """
+    Create the package description, package directory and package.xml file.
 
-    output_file_path = os.path.join(output_directory, output_file_name)
+    :param args: The commandline args, a package_create_args object.
+    :param buildtool_depends: List of build_tool dependencies.
+    :param test_dependencies: List of test dependencies.
+    :return package: New package details.
+    :return package_directory: New directory for the package.
+    :return error_message: Describes any error that occurs, None if no errors occured.
+    """
+    package = create_package(args, buildtool_depends, test_dependencies)
 
-    print('creating', output_file_path)
-    _expand_template(template_path, template_config, output_file_path)
+    package_path = os.path.join(args.destination_directory, package.name)
+    result, error_message = valid_package_directory(package_path)
+
+    if not result:
+        return None, None, error_message
+
+    display_create_package_environment_message(args, package)
+    package_directory = create_package_environment(package, args.destination_directory)
+
+    if not package_directory:
+        error_message = 'Unable to create folder: ' + package_directory
+
+    return package, package_directory, error_message
+
+
+def create_package(args, buildtool_depends=None, test_dependencies=[]):
+    """
+    Create a package description object.
+
+    :param args: Commandline args wrapped by package_create_arguments object.
+    :param buildtool_depends: List of buildtool dependencies.
+    :param test_dependencies: List of test dependencies.
+    :return: New package.
+    """
+    if not buildtool_depends:
+        buildtool_depends = args.build_type
```

Comment:
This line results in the bug that creates multiple <buildtool_depend> entries in package.xml as reported here https://github.com/ros2/ros2cli/pull/444#pullrequestreview-377426003.  The correct code should be:
```
buildtool_depends = [args.build_type]
```